sdks/python/apache_beam/transforms/dataflow_distribution_counter_test.py

```diff
@@ -9,24 +9,27 @@
 # limitations under the License.
 #
 
-'''
-Unit tests for the Distribution Counter
-'''
+"""Unit tests for DataflowDistributionCounter
+When Cython is available, unit tests will test on cythonized module,
+otherwise, test on pure python module
+"""
+
 import math
 import unittest
 
-from apache_beam.transforms.cy_combiners import DistributionAccumulator
+from mock import Mock
 
+from apache_beam.transforms import DataflowDistributionCounter
 
-class DistributionAccumulatorTest(unittest.TestCase):
 
+class DataflowDistributionAccumulatorTest(unittest.TestCase):
   def test_calculate_bucket_index_with_input_0(self):
-    counter = DistributionAccumulator()
+    counter = DataflowDistributionCounter()
```

Comment:
<!--new_thread; commit:89d005daa5313826869ebdad0aba047ae95febcb; resolved:0-->
Can we do the following, to get extra test coverage, for each of these test methods?  Can you create a new method on this class, say, `create_distribution_counters()` that does, for example:

```python
def create_distribution_counters(self):
  counters = [SlowDataflowDistributionCounter()]
  if DataflowDistributionCounter != SlowDataflowDistributionCounter:
    # Fast Cythonized distribution counter available.
    counters.append(DataflowDistributionCounter())
  return counters
```

Then, you can wrap each of the tests here like this:

```python
for counter in self.create_distribution_counters():
  index = counter.calculate_bucket_index(0)
  self.assertEquals(index, 0)
  # ...etc.
```

This way, we get test coverage for all available distribution counter implemenations.

Reply:
There is a import pattern in transforms/_init_.py (https://github.com/apache/beam/pull/5028/files/dd3ae5364c173a99c8a15f7e80960b9307993f7c#diff-a827371b61aef78df1df61f6d34c0716R28). So if running tox -e py27, this unit tests will test on python module, if running tox -2 py27-cython, tests will test on cython module.

Reply:
Thank you for pointing out that we run both python-only and cython tests.  Please ignore this comment in that case.